### ISSUE TITLE
Upgrading 1.13.1 to 1.13.2 - panic: assignment to entry in nil map

### ISSUE THREAD

**Issue by corybolar** [cilium/cilium#24975]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Cilium fails to start after upgrading to 1.13.2 from 1.13.1 or 1.13.0.  No other changes and is reproducible every time with my config.

Cluster makes use of the bgp control plane.  Pasting the contents of the CiliumBGPPeeringPolicies
```
apiVersion: cilium.io/v2alpha1
kind: CiliumBGPPeeringPolicy
metadata:
  name: 01-bgp-peering-policy
spec:
  nodeSelector:
    matchLabels:
      kubernetes.io/os: linux
  virtualRouters:
  - exportPodCIDR: true
    localASN: 65006
    neighbors:
    - peerASN: 65001
      peerAddress: 192.168.48.3/32
    - peerASN: 65001
      peerAddress: fd9c:de81:5c4d:8::1/128
    serviceSelector:
      matchExpressions:
      - key: somekey
        operator: NotIn
        values:
        - never-used-value
```

### Cilium Version

1.13.2

1.13.1 and 1.13.0 allow cilium to start successfully.  Installation and upgrade are performed via helm.

### Kernel Version

Linux 5.14.0-162.23.1.el9_1.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Mar 23 20:08:28 EDT 2023 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

Server Version: v1.24.12+rke2r1

### Sysdump

N/a.  Cilium will not run

### Relevant log output

```shell
config Running
config level=info msg=Invoked duration="484.224µs" function="cmd.glob..func35 (build-config.go:32)" subsys=hive
config level=info msg="Establishing connection to apiserver" host="https://kube.brke2.cbtek.io:6443" subsys=k8s-client
config level=info msg="Connected to apiserver" subsys=k8s-client
config level=info msg="Start hook executed" duration=8.568422ms function="client.(*compositeClientset).onStart" subsys=hive
config level=info msg="Reading configuration from config-map:kube-system/cilium-config" configSource="config-map:kube-system/cilium-config" subsys=option-resolver
config level=info msg="Got 100 config pairs from source" configSource="config-map:kube-system/cilium-config" subsys=option-resolver
config level=info msg="Reading configuration from cilium-node-config:kube-system/" configSource="cilium-node-config:kube-system/" subsys=option-resolver
config level=info msg="Got 0 config pairs from source" configSource="cilium-node-config:kube-system/" subsys=option-resolver
config level=info msg="Start hook executed" duration=6.187137ms function="cmd.(*buildConfig).onStart" subsys=hive
config level=info msg="Stop hook executed" duration="8.336µs" function="client.(*compositeClientset).onStop" subsys=hive
mount-bpf-fs none on /sys/fs/bpf type bpf (rw,nosuid,nodev,noexec,relatime,mode=700)
apply-sysctl-overwrites sysctl config up-to-date, nothing to do
install-cni-binaries Installing cilium-cni to /host/opt/cni/bin/ ...
install-cni-binaries wrote /host/opt/cni/bin/cilium-cni
cilium-agent level=info msg="  --pprof-port='6060'" subsys=daemon
cilium-agent level=info msg="  --preallocate-bpf-maps='false'" subsys=daemon
cilium-agent level=info msg="  --prepend-iptables-chains='true'" subsys=daemon
cilium-agent level=info msg="  --procfs='/host/proc'" subsys=daemon
cilium-agent level=info msg="  --prometheus-serve-addr=':19090'" subsys=daemon
cilium-agent level=info msg="  --proxy-connect-timeout='1'" subsys=daemon
cilium-agent level=info msg="  --proxy-gid='1337'" subsys=daemon
cilium-agent level=info msg="  --proxy-max-connection-duration-seconds='0'" subsys=daemon
cilium-agent level=info msg="  --proxy-max-requests-per-connection='0'" subsys=daemon
cilium-agent level=info msg="  --proxy-prometheus-port='9964'" subsys=daemon
cilium-agent level=info msg="  --read-cni-conf=''" subsys=daemon
cilium-agent level=info msg="  --remove-cilium-node-taints='true'" subsys=daemon
cilium-agent level=info msg="  --restore='true'" subsys=daemon
cilium-agent level=info msg="  --route-metric='0'" subsys=daemon
cilium-agent level=info msg="  --set-cilium-is-up-condition='true'" subsys=daemon
cilium-agent level=info msg="  --sidecar-istio-proxy-image='cilium/istio_proxy'" subsys=daemon
cilium-agent level=info msg="  --single-cluster-route='false'" subsys=daemon
mount-cgroup level=info msg="Mounted cgroupv2 filesystem at /run/cilium/cgroupv2" subsys=cgroups
cilium-agent level=info msg="  --skip-cnp-status-startup-clean='false'" subsys=daemon
cilium-agent level=info msg="  --socket-path='/var/run/cilium/cilium.sock'" subsys=daemon
cilium-agent level=info msg="  --sockops-enable='false'" subsys=daemon
cilium-agent level=info msg="  --srv6-encap-mode='reduced'" subsys=daemon
cilium-agent level=info msg="  --state-dir='/var/run/cilium'" subsys=daemon
cilium-agent level=info msg="  --synchronize-k8s-nodes='true'" subsys=daemon
cilium-agent level=info msg="  --tofqdns-dns-reject-response-code='refused'" subsys=daemon
cilium-agent level=info msg="  --tofqdns-enable-dns-compression='true'" subsys=daemon
cilium-agent level=info msg="  --tofqdns-endpoint-max-ip-per-hostname='50'" subsys=daemon
cilium-agent level=info msg="  --tofqdns-idle-connection-grace-period='0s'" subsys=daemon
cilium-agent level=info msg="  --tofqdns-max-deferred-connection-deletes='10000'" subsys=daemon
cilium-agent level=info msg="  --tofqdns-min-ttl='3600'" subsys=daemon
cilium-agent level=info msg="  --tofqdns-pre-cache=''" subsys=daemon
cilium-agent level=info msg="  --tofqdns-proxy-port='0'" subsys=daemon
cilium-agent level=info msg="  --tofqdns-proxy-response-max-delay='100ms'" subsys=daemon
cilium-agent level=info msg="  --trace-payloadlen='128'" subsys=daemon
cilium-agent level=info msg="  --trace-sock='true'" subsys=daemon
cilium-agent level=info msg="  --tunnel='disabled'" subsys=daemon
cilium-agent level=info msg="  --tunnel-port='0'" subsys=daemon
cilium-agent level=info msg="  --unmanaged-pod-watcher-interval='15'" subsys=daemon
cilium-agent level=info msg="  --version='false'" subsys=daemon
cilium-agent level=info msg="  --vlan-bpf-bypass=''" subsys=daemon
cilium-agent level=info msg="  --vtep-cidr=''" subsys=daemon
cilium-agent level=info msg="  --vtep-endpoint=''" subsys=daemon
cilium-agent level=info msg="  --vtep-mac=''" subsys=daemon
cilium-agent level=info msg="  --vtep-mask=''" subsys=daemon
cilium-agent level=info msg="  --write-cni-conf-when-ready=''" subsys=daemon
cilium-agent level=info msg="     _ _ _" subsys=daemon
cilium-agent level=info msg=" ___|_| |_|_ _ _____" subsys=daemon
cilium-agent level=info msg="|  _| | | | | |     |" subsys=daemon
cilium-agent level=info msg="|___|_|_|_|___|_|_|_|" subsys=daemon
cilium-agent level=info msg="Cilium 1.13.2 8cb94c70 2023-04-17T23:19:21+02:00 go version go1.19.8 linux/amd64" subsys=daemon
cilium-agent level=info msg="Envoy version check disabled" subsys=daemon
cilium-agent level=info msg="clang (10.0.0) and kernel (5.14.0) versions: OK!" subsys=linux-datapath
cilium-agent level=info msg="linking environment: OK!" subsys=linux-datapath
cilium-agent level=info msg="Kernel config file not found: if the agent fails to start, check the system requirements at https://docs.cilium.io/en/stable/operations/system_requirements" subsys=probes
cilium-agent level=info msg="Detected mounted BPF filesystem at /sys/fs/bpf" subsys=bpf
cilium-agent level=info msg="Mounted cgroupv2 filesystem at /run/cilium/cgroupv2" subsys=cgroups
cilium-agent level=info msg="Parsing base label prefixes from default label list" subsys=labels-filter
cilium-agent level=info msg="Parsing additional label prefixes from user inputs: []" subsys=labels-filter
cilium-agent level=info msg="Final label prefixes to be used for identity evaluation:" subsys=labels-filter
cilium-agent level=info msg=" - reserved:.*" subsys=labels-filter
cilium-agent level=info msg=" - :io\\.kubernetes\\.pod\\.namespace" subsys=labels-filter
cilium-agent level=info msg=" - :io\\.cilium\\.k8s\\.namespace\\.labels" subsys=labels-filter
cilium-agent level=info msg=" - :app\\.kubernetes\\.io" subsys=labels-filter
cilium-agent level=info msg=" - !:io\\.kubernetes" subsys=labels-filter
cilium-agent level=info msg=" - !:kubernetes\\.io" subsys=labels-filter
cilium-agent level=info msg=" - !:.*beta\\.kubernetes\\.io" subsys=labels-filter
cilium-agent level=info msg=" - !:k8s\\.io" subsys=labels-filter
cilium-agent level=info msg=" - !:pod-template-generation" subsys=labels-filter
cilium-agent level=info msg=" - !:pod-template-hash" subsys=labels-filter
cilium-agent level=info msg=" - !:controller-revision-hash" subsys=labels-filter
cilium-agent level=info msg=" - !:annotation.*" subsys=labels-filter
cilium-agent level=info msg=" - !:etcd_node" subsys=labels-filter
cilium-agent level=info msg="Using autogenerated IPv4 allocation range" subsys=node v4Prefix=10.2.0.0/16
cilium-agent level=info msg="Using autogenerated IPv6 allocation range" subsys=node v6Prefix="f00d::a02:0:0:0/96"
cilium-agent level=info msg=Invoked duration="483.314µs" function="gops.registerGopsHooks (cell.go:39)" subsys=hive
cilium-agent level=info msg=Invoked duration=2.622952ms function="cmd.glob..func2 (daemon_main.go:1644)" subsys=hive
cilium-agent level=info msg="Started gops server" address="127.0.0.1:9890" subsys=gops
cilium-agent level=info msg="Start hook executed" duration="275.771µs" function="gops.registerGopsHooks.func1 (cell.go:44)" subsys=hive
cilium-agent level=info msg="Establishing connection to apiserver" host="https://kube.brke2.cbtek.io:6443" subsys=k8s-client
cilium-agent level=info msg="Connected to apiserver" subsys=k8s-client
cilium-agent level=info msg="Start hook executed" duration=6.89865ms function="client.(*compositeClientset).onStart" subsys=hive
cilium-agent level=info msg="Start hook executed" duration=8.633767ms function="cmd.newDatapath.func1 (daemon_main.go:1624)" subsys=hive
cilium-agent level=info msg="Start hook executed" duration="3.897µs" function="*resource.resource[*github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1.Service].Start" subsys=hive
cilium-agent level=info msg="Start hook executed" duration=100.885769ms function="*gobgp.diffStore[*github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1.Service].Start" subsys=hive
cilium-agent level=info msg="Start hook executed" duration="1.974µs" function="*resource.resource[*github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1.CiliumBGPPeeringPolicy].Start" subsys=hive
cilium-agent level=info msg="Start hook executed" duration="1.616µs" function="*resource.resource[*k8s.io/api/core/v1.Node].Start" subsys=hive
cilium-agent level=info msg="Start hook executed" duration=781ns function="*resource.resource[*github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2.CiliumNode].Start" subsys=hive
cilium-agent level=info msg="Start hook executed" duration="52.762µs" function="*agent.ciliumNodeSpecer.Start" subsys=hive
cilium-agent level=info msg="Start hook executed" duration=101.040788ms function="*agent.Controller.Start" subsys=hive
cilium-agent level=info msg="Cilium BGP Control Plane Controller now running..." component=Controller.Run subsys=bgp-control-plane
cilium-agent level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
cilium-agent level=info msg="Registering GoBGP servers for policy with local ASN 65006" component=gobgp.RouterManager.registerBGPServer subsys=bgp-control-plane
cilium-agent level=info msg="Reconciling peers for virtual router with local ASN 65006" component=gobgp.neighborReconciler subsys=bgp-control-plane
cilium-agent level=info msg="Adding peer 192.168.48.3/32 65001 to local ASN 65006" component=gobgp.neighborReconciler subsys=bgp-control-plane
cilium-agent level=info msg="Add a peer configuration" Key=192.168.48.3 Topic=Peer asn=65006 component=gobgp.BgpServerInstance subsys=bgp-control-plane
cilium-agent level=info msg="Adding peer fd9c:de81:5c4d:8::1/128 65001 to local ASN 65006" component=gobgp.neighborReconciler subsys=bgp-control-plane
cilium-agent level=info msg="Using Managed Neighbor Kernel support" subsys=daemon
cilium-agent level=info msg="Auto-enabling \"enable-node-port\", \"enable-external-ips\", \"bpf-lb-sock\", \"enable-host-port\", \"enable-session-affinity\" features" subsys=daemon
cilium-agent level=info msg="Add a peer configuration" Key="fd9c:de81:5c4d:8::1" Topic=Peer asn=65006 component=gobgp.BgpServerInstance subsys=bgp-control-plane
cilium-agent level=info msg="Done reconciling peers for virtual router with local ASN 65006" component=gobgp.neighborReconciler subsys=bgp-control-plane
cilium-agent level=info msg="type:STATE  peer:{conf:{local_asn:65006  neighbor_address:\"192.168.48.3\"  peer_asn:65001}  state:{local_asn:65006  neighbor_address:\"192.168.48.3\"  peer_asn:65001  session_state:IDLE  router_id:\"<nil>\"}  transport:{local_address:\"<nil>\"}}"
cilium-agent level=info msg="type:STATE  peer:{conf:{local_asn:65006  neighbor_address:\"fd9c:de81:5c4d:8::1\"  peer_asn:65001}  state:{local_asn:65006  neighbor_address:\"fd9c:de81:5c4d:8::1\"  peer_asn:65001  session_state:IDLE  router_id:\"<nil>\"}  transport:{local_address:\"<nil>\"}}"
cilium-agent level=info msg="Successfully registered GoBGP servers for policy with local ASN 65006" component=gobgp.RouterManager.registerBGPServer subsys=bgp-control-plane
cilium-agent level=info msg="type:STATE  peer:{conf:{local_asn:65006  neighbor_address:\"192.168.48.3\"  peer_asn:65001}  state:{local_asn:65006  neighbor_address:\"192.168.48.3\"  peer_asn:65001  session_state:ACTIVE  router_id:\"<nil>\"}  transport:{local_address:\"<nil>\"}}"
cilium-agent level=info msg="type:STATE  peer:{conf:{local_asn:65006  neighbor_address:\"fd9c:de81:5c4d:8::1\"  peer_asn:65001}  state:{local_asn:65006  neighbor_address:\"fd9c:de81:5c4d:8::1\"  peer_asn:65001  session_state:ACTIVE  router_id:\"<nil>\"}  transport:{local_address:\"<nil>\"}}"
cilium-agent level=info msg="Inheriting MTU from external network interface" device=wg0 ipAddr=172.24.0.2 mtu=1420 subsys=mtu
cilium-agent level=info msg="L7 proxies are disabled" subsys=daemon
cilium-agent level=info msg="Cgroup metadata manager is enabled" subsys=cgroup-manager
cilium-agent level=info msg="Restored 5 node IDs from the BPF map" subsys=linux-datapath
cilium-agent level=info msg="Restored services from maps" failedServices=0 restoredServices=266 subsys=service
cilium-agent level=info msg="Restored backends from maps" failedBackends=0 restoredBackends=169 subsys=service
cilium-agent level=info msg="Reading old endpoints..." subsys=daemon
cilium-agent level=info msg="Waiting until all Cilium CRDs are available" subsys=k8s
cilium-agent level=info msg="All Cilium CRDs have been found and are available" subsys=k8s
cilium-agent level=info msg="Creating or updating CiliumNode resource" node=bworker01 subsys=nodediscovery
cilium-agent level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
cilium-agent level=info msg="Done reconciling peers for virtual router with local ASN 65006" component=gobgp.neighborReconciler subsys=bgp-control-plane
cilium-agent level=info msg="Retrieved node information from cilium node" nodeName=bworker01 subsys=k8s
cilium-agent level=info msg="Received own node information from API server" ipAddr.ipv4=192.168.48.141 ipAddr.ipv6="<nil>" k8sNodeIP=192.168.48.141 labels="map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux kubernetes.io/arch:amd64 kubernetes.io/hostname:bworker01 kubernetes.io/os:linux]" nodeName=bworker01 subsys=k8s v4Prefix=10.42.1.0/24 v6Prefix="fd9c:de81:5c4d:31:4000:0:1:0/112"
cilium-agent level=info msg="k8s mode: Allowing localhost to reach local endpoints" subsys=daemon
cilium-agent level=info msg="Direct routing device detected" direct-routing-device=eth0 subsys=linux-datapath
cilium-agent level=info msg="Detected devices" devices="[eth0]" subsys=linux-datapath
cilium-agent level=info msg="BPF host routing is currently not compatible with Wireguard (--enable-wireguard). Falling back to legacy host routing (enable-host-legacy-routing=true)." subsys=daemon
cilium-agent level=info msg="Masquerading IP selected for device" device=eth0 ipv4=192.168.48.141 subsys=node
cilium-agent level=info msg="Enabling k8s event listener" subsys=k8s-watcher
Stream closed EOF for kube-system/cilium-5xfn4 (config)
cilium-agent level=info msg="Removing stale endpoint interfaces" subsys=daemon
cilium-agent level=info msg="Waiting until local node addressing before starting watchers depending on it" subsys=k8s-watcher
cilium-agent level=info msg="Skipping kvstore configuration" subsys=daemon
cilium-agent level=info msg="Restored router address from node_config" file=/var/run/cilium/state/globals/node_config.h ipv4=10.42.1.90 ipv6="fd9c:de81:5c4d:31:4000:0:1:ecef" subsys=node
cilium-agent level=info msg="Initializing node addressing" subsys=daemon
cilium-agent level=info msg="Initializing cluster-pool IPAM" subsys=ipam v4Prefix=10.42.1.0/24 v6Prefix="fd9c:de81:5c4d:31:4000:0:1:0/112"
cilium-agent level=info msg="Restoring endpoints..." subsys=daemon
cilium-agent level=info msg="Endpoints restored" failed=0 restored=42 subsys=daemon
cilium-agent level=info msg="Addressing information:" subsys=daemon
cilium-agent level=info msg="  Cluster-Name: default" subsys=daemon
cilium-agent level=info msg="  Cluster-ID: 0" subsys=daemon
cilium-agent level=info msg="  Local node-name: bworker01" subsys=daemon
cilium-agent level=info msg="  Node-IPv6: 2600:4040:2088:7430::1000:167d" subsys=daemon
cilium-agent level=info msg="  IPv6 allocation prefix: fd9c:de81:5c4d:31:4000:0:1:0/112" subsys=daemon
cilium-agent level=info msg="  IPv6 native routing prefix: fd9c:de81:5c4d:30::1000:0/112" subsys=daemon
cilium-agent level=info msg="  IPv6 router address: fd9c:de81:5c4d:31:4000:0:1:ecef" subsys=daemon
cilium-agent level=info msg="  Local IPv6 addresses:" subsys=daemon
cilium-agent level=info msg="  - fd9c:de81:5c4d:30::1000:167d" subsys=daemon
cilium-agent level=info msg="  - 2600:4040:2088:7430::1000:167d" subsys=daemon
cilium-agent level=info msg="  - 2600:4040:2088:7430:5054:ff:fe5d:de1f" subsys=daemon
cilium-agent level=info msg="  - fd9c:de81:5c4d:30:5054:ff:fe5d:de1f" subsys=daemon
cilium-agent level=info msg="  External-Node IPv4: 192.168.48.141" subsys=daemon
cilium-agent level=info msg="  Internal-Node IPv4: 10.42.1.90" subsys=daemon
cilium-agent level=info msg="  IPv4 allocation prefix: 10.42.1.0/24" subsys=daemon
cilium-agent level=info msg="  IPv4 native routing prefix: 192.168.48.128/26" subsys=daemon
cilium-agent level=info msg="  Loopback IPv4: 169.254.42.1" subsys=daemon
cilium-agent level=info msg="  Local IPv4 addresses:" subsys=daemon
cilium-agent level=info msg="  - 192.168.48.141" subsys=daemon
cilium-agent level=info msg="  - 172.24.0.2" subsys=daemon
cilium-agent level=info msg="Adding local node to cluster" node="{bworker01 default [{InternalIP 192.168.48.141} {InternalIP 2600:4040:2088:7430::1000:167d} {CiliumInternalIP 10.42.1.90} {CiliumInternalIP fd9c:de81:5c4d:31:4000:0:1:ecef}] 10.42.1.0/24 [] fd9c:de81:5c4d:31:4000:0:1:0/112 [] 10.42.1.13 fd9c:de81:5c4d:31:4000:0:1:eb46 <nil> <nil> 0 local 0 map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux kubernetes.io/arch:amd64 kubernetes.io/hostname:bworker01 kubernetes.io/os:linux] map[] 1 /IRWLLOBe9SglFfmRtubSDMhuJ89J4w5o+w03oEyNl8=}" subsys=nodediscovery
cilium-agent level=info msg="Creating or updating CiliumNode resource" node=bworker01 subsys=nodediscovery
cilium-agent level=info msg="Waiting until all pre-existing resources have been received" subsys=k8s-watcher
cilium-agent level=info msg="Annotating k8s node" subsys=daemon v4CiliumHostIP.IPv4=10.42.1.90 v4IngressIP.IPv4="<nil>" v4Prefix=10.42.1.0/24 v4healthIP.IPv4=10.42.1.13 v6CiliumHostIP.IPv6="fd9c:de81:5c4d:31:4000:0:1:ecef" v6IngressIP.IPv6="<nil>" v6Prefix="fd9c:de81:5c4d:31:4000:0:1:0/112" v6healthIP.IPv6="fd9c:de81:5c4d:31:4000:0:1:eb46"
cilium-agent level=info msg="Initializing identity allocator" subsys=identity-cache
cilium-agent level=info msg="Allocating identities between range" cluster-id=0 max=65535 min=256 subsys=identity-cache
cilium-agent level=info msg="Cluster-ID is not specified, skipping ClusterMesh initialization" subsys=daemon
cilium-agent level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
cilium-agent level=info msg="Virtual router with ASN 65006 router ID has changed from 172.24.0.2 to 192.168.48.141" component=gobgp.preflightReconciler subsys=bgp-control-plane
cilium-agent level=info msg="Recreating virtual router with ASN 65006 for changes to take effect" component=gobgp.preflightReconciler subsys=bgp-control-plane
cilium-agent level=info msg="Delete a peer configuration" Key=192.168.48.3 Topic=Peer asn=65006 component=gobgp.BgpServerInstance subsys=bgp-control-plane
cilium-agent level=info msg="Delete a peer configuration" Key="fd9c:de81:5c4d:8::1" Topic=Peer asn=65006 component=gobgp.BgpServerInstance subsys=bgp-control-plane
cilium-agent level=info msg="Reconciling peers for virtual router with local ASN 65006" component=gobgp.neighborReconciler subsys=bgp-control-plane
cilium-agent level=info msg="Adding peer 192.168.48.3/32 65001 to local ASN 65006" component=gobgp.neighborReconciler subsys=bgp-control-plane
cilium-agent level=info msg="Add a peer configuration" Key=192.168.48.3 Topic=Peer asn=65006 component=gobgp.BgpServerInstance subsys=bgp-control-plane
cilium-agent level=info msg="Adding peer fd9c:de81:5c4d:8::1/128 65001 to local ASN 65006" component=gobgp.neighborReconciler subsys=bgp-control-plane
cilium-agent level=info msg="Add a peer configuration" Key="fd9c:de81:5c4d:8::1" Topic=Peer asn=65006 component=gobgp.BgpServerInstance subsys=bgp-control-plane
cilium-agent level=info msg="type:STATE  peer:{conf:{local_asn:65006  neighbor_address:\"192.168.48.3\"  peer_asn:65001}  state:{local_asn:65006  neighbor_address:\"192.168.48.3\"  peer_asn:65001  session_state:IDLE  router_id:\"<nil>\"}  transport:{local_address:\"<nil>\"}}"
cilium-agent level=info msg="type:STATE  peer:{conf:{local_asn:65006  neighbor_address:\"fd9c:de81:5c4d:8::1\"  peer_asn:65001}  state:{local_asn:65006  neighbor_address:\"fd9c:de81:5c4d:8::1\"  peer_asn:65001  session_state:IDLE  router_id:\"<nil>\"}  transport:{local_address:\"<nil>\"}}"
cilium-agent level=info msg="type:STATE  peer:{conf:{local_asn:65006  neighbor_address:\"192.168.48.3\"  peer_asn:65001}  state:{local_asn:65006  neighbor_address:\"192.168.48.3\"  peer_asn:65001  session_state:ACTIVE  router_id:\"<nil>\"}  transport:{local_address:\"<nil>\"}}"
cilium-agent level=info msg="Done reconciling peers for virtual router with local ASN 65006" component=gobgp.neighborReconciler subsys=bgp-control-plane
cilium-agent level=info msg="type:STATE  peer:{conf:{local_asn:65006  neighbor_address:\"fd9c:de81:5c4d:8::1\"  peer_asn:65001}  state:{local_asn:65006  neighbor_address:\"fd9c:de81:5c4d:8::1\"  peer_asn:65001  session_state:ACTIVE  router_id:\"<nil>\"}  transport:{local_address:\"<nil>\"}}"
cilium-agent panic: assignment to entry in nil map
cilium-agent 
cilium-agent goroutine 233 [running]:
cilium-agent github.com/cilium/cilium/pkg/bgpv1/gobgp.(*LBServiceReconciler).reconcileService(0x0?, {0x3888e58, 0xc000b72000}, 0xc000de7e00, 0x9?, 0x0?)
cilium-agent     /go/src/github.com/cilium/cilium/pkg/bgpv1/gobgp/reconcile.go:563 +0x38e
cilium-agent github.com/cilium/cilium/pkg/bgpv1/gobgp.(*LBServiceReconciler).fullReconciliation(0xc000d69cb0, {0x3888e58, 0xc000b72000}, 0x2?, 0xc000de7e00, 0xc00180cf40?, 0x1f8e235?)
cilium-agent     /go/src/github.com/cilium/cilium/pkg/bgpv1/gobgp/reconcile.go:467 +0x29f
cilium-agent github.com/cilium/cilium/pkg/bgpv1/gobgp.(*LBServiceReconciler).Reconcile(0xc0014fa390?, {0x3888e58?, 0xc000b72000?}, 0x2bb07c0?, 0x4137d7?, 0x0?, 0x0?)
cilium-agent     /go/src/github.com/cilium/cilium/pkg/bgpv1/gobgp/reconcile.go:419 +0xf3
cilium-agent github.com/cilium/cilium/pkg/bgpv1/gobgp.(*BGPRouterManager).reconcileBGPConfig(0xc000d75f00, {0x3888e58, 0xc000b72000}, 0xc000de7e00, 0xc001214000, 0x1?)
cilium-agent     /go/src/github.com/cilium/cilium/pkg/bgpv1/gobgp/routermgr.go:357 +0x14c
cilium-agent github.com/cilium/cilium/pkg/bgpv1/gobgp.(*BGPRouterManager).reconcile(0xc000d75f00, {0x3888e58, 0xc000b72000}, 0xc00180d460)
cilium-agent     /go/src/github.com/cilium/cilium/pkg/bgpv1/gobgp/routermgr.go:325 +0x205
cilium-agent github.com/cilium/cilium/pkg/bgpv1/gobgp.(*BGPRouterManager).ConfigurePeers(0xc000d75f00, {0x3888e58, 0xc000b72000}, 0xc000f71400, 0xc0005ccb90)
cilium-agent     /go/src/github.com/cilium/cilium/pkg/bgpv1/gobgp/routermgr.go:153 +0x59b
cilium-agent github.com/cilium/cilium/pkg/bgpv1/agent.(*Controller).Reconcile(0xc000393440, {0x3888e58, 0xc000b72000})
cilium-agent     /go/src/github.com/cilium/cilium/pkg/bgpv1/agent/controller.go:389 +0x76d
cilium-agent github.com/cilium/cilium/pkg/bgpv1/agent.(*Controller).Run(0xc000393440, {0x3888e58, 0xc000b72000})
cilium-agent     /go/src/github.com/cilium/cilium/pkg/bgpv1/agent/controller.go:259 +0x279
cilium-agent github.com/cilium/cilium/pkg/bgpv1/agent.(*Controller).Start.func3({0x3888e58?, 0xc000b72000?})
cilium-agent     /go/src/github.com/cilium/cilium/pkg/bgpv1/agent/controller.go:201 +0x30
cilium-agent github.com/cilium/workerpool.(*WorkerPool).run.func1()
cilium-agent     /go/src/github.com/cilium/cilium/vendor/github.com/cilium/workerpool/workerpool.go:174 +0x82
cilium-agent created by github.com/cilium/workerpool.(*WorkerPool).run
cilium-agent     /go/src/github.com/cilium/cilium/vendor/github.com/cilium/workerpool/workerpool.go:171 +0x4e
```

### Anything else?

Upgrade to 1.13.2 is important to me as it fixes https://github.com/cilium/cilium/issues/24069

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
